remote/remote_test.go

```diff
@@ -1595,5 +1658,47 @@ func testImage(t *testing.T, when spec.G, it spec.S) {
 				h.AssertEq(t, image.CheckReadWriteAccess(), false)
 			})
 		})
+
+		when("using custom handler", func() {
+			it.Before(func() {
+				os.Setenv("DOCKER_CONFIG", customRegistry.DockerDirectory)
+			})
+
+			it.After(func() {
+				os.Setenv("DOCKER_CONFIG", dockerRegistry.DockerDirectory)
+			})
+
+			when("image has read/write access", func() {
+				it("returns true", func() {
+					image, err := remote.NewImage(customRegistry.RepoName(readWriteImage), authn.DefaultKeychain)
+					h.AssertNil(t, err)
+					h.AssertEq(t, image.CheckReadWriteAccess(), true)
+				})
+			})
+
+			when("image has read access but no write access", func() {
+				it("returns false", func() {
+					image, err := remote.NewImage(customRegistry.RepoName(readNoWriteImage), authn.DefaultKeychain)
+					h.AssertNil(t, err)
+					h.AssertEq(t, image.CheckReadWriteAccess(), false)
```

Comment:
Should we check read access here?

Reply:
Do you mean adding the assert `image.CheckReadAccess()`? because that one seems to be tested at line 1585

Reply:
Ohh I missed that these were all scoped to `CheckReadAccess()`. I think it's okay.